### PR TITLE
Document installation and getting started tutorial for provision

### PR DESCRIPTION
**Pull request by cfchase**
Added installation instructions to README.md
Added first commit of getting_started.md
Added provision instructions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,19 +1,48 @@
 # Ansible Playbook Bundle (APB)
 
-Ansible Playbook Bundle (APB) is a lightweight application definition (meta-containers). APB
-has the following features:
+An **Ansible Playbook Bundle (APB)** is a lightweight application definition (meta-container). They are used to define and deploy complex groups of applications, deployment configs, deployments, and services to an [OpenShift Origin](https://github.com/OpenShift/origin) cluster running the [Ansible Service Broker](https://github.com/fusor/ansible-service-broker).  APBs offer more power and  simple configuration by leveraging the power of Ansible. APBs have the following features:
```
Comment (jmrodri) [inline README.md]: extra space between 'and simple'.

Comment (jmrodri) [reply README.md]: change 'power of Ansible' to be a link to ansible `power of [Ansible](http://www.ansible.com)`

File: README.md [hunk 2]
```diff
@@ -1,19 +1,48 @@
...
 
 * Metadata contains list of required/optional parameters for use during deployment.
 * Leverages existing investment in Ansible Roles / Playbooks.
-* Simple directory with named “action” playbooks and metadata.
-* Developer Tooling to drive guided approach.
+* Actions under a directory with named "playbooks" and metadata defined in "apb.yml".
```
Comment (jmrodri) [inline README.md]: I would format the words in quotes with either *italics*, **bold** or **_both_**.

File: README.md [hunk 3]
```diff
@@ -1,19 +1,48 @@
...
+## Installing the ***apb*** tool
+##### Prerequisites
+Install Docker
+
+##### RPM Installation
+Download from [here](????)
+Install using dnf/yum
+```bash
+
+```
```
Comment (jmrodri) [inline README.md]: What does this do?

File: README.md [hunk 4]
```diff
@@ -1,19 +1,48 @@
...
+
+##### Installing from source
+Clone this repo
+```bash
+git clone https://github.com/fusor/ansible-playbook-bundle.git
+```
+Install requirements and run the setup script (requires python)
+```bash
+sudo pip install -r src/requirements.txt && sudo python setup.py install
+```
```
Comment (jmrodri) [inline README.md]: I'm much more partial to using tito to install :)
```
tito build --rpm -i
```

Or make it:

##### Installing from source
Clone this repo
```bash
git clone https://github.com/fusor/ansible-playbook-bundle.git
```
Install requirements and run the setup script (requires python)
```bash
sudo pip install -r src/requirements.txt && sudo python setup.py install
```
Alternatively you can use [tito](http://github.com/dgoodwin/tito) to install.
```bash
tito build --test --rpm -i
```

File: README.md [hunk 5]
```diff
@@ -1,19 +1,48 @@
...
+
+##### Test apb tooling
+```bash
+apb help
+```
+
+<a name="docs"></a>
```
Comment (jmrodri) [inline README.md]: I thought markdown creates the links automagically.

File: docs/getting_started.md [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+    2. [Actions](#actions)
+        1. [Provision](#provision)
+        2. [Bind](#bind)
+        3. [Unbind](#unbind)
+        4. [Deprovision](#deprovision)
+        
+4. [More information](#more-info)
+
+<a name="intro"></a>
+## Introduction Ansible Playbook Bundles (APBs)
```
Comment (jmrodri) [inline docs/getting_started.md]: Introduction to Ansible PlaybookBundles (APBs)?

File: docs/getting_started.md [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+        2. [Bind](#bind)
+        3. [Unbind](#unbind)
+        4. [Deprovision](#deprovision)
+        
+4. [More information](#more-info)
+
+<a name="intro"></a>
+## Introduction Ansible Playbook Bundles (APBs)  
+ 
+In this tutorial, we'll walk through the creation of a APB for a PostgreSQL Database.  You will be able to provision, deprovision, and bind or unbind it to other applications.  You can find more information about the design of APBs in the [design doc](https://github.com/fusor/ansible-playbook-bundle/blob/master/docs/design.md)
```
Comment (jmrodri) [inline docs/getting_started.md]: an APB?

Comment (jmrodri) [reply docs/getting_started.md]: database instead of Database.

File: docs/getting_started.md [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+  
+  
+<a name="prep-dev"></a>
+## Prepare your development environment 
+Install APB tooling as documented in the main [README](https://github.com/cfchase/ansible-playbook-bundle/tree/docs#install)
```
Comment (jmrodri) [inline docs/getting_started.md]: I'd remove main. '... in the [README]'

Comment (jmrodri) [reply docs/getting_started.md]: The link says cfchase.

File: docs/getting_started.md [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+
+Create a local development environment with both a [Service Catalog](https://github.com/kubernetes-incubator/service-catalog) and [Ansible Service Broker](https://github.com/fusor/ansible-service-broker).  We will do this using [catasb](https://github.com/fusor/catasb), a collection of scripts which use Ansible to automate the set up of the cluster for you on a local host, ec2, or virtual machines.  For this tutorial we'll be assuming the locally hosted environment which is documented at [https://github.com/fusor/catasb/blob/master/local/README.md](https://github.com/fusor/catasb/blob/master/local/README.md).  After completing the set up, take note of the OpenShift cluster **host:port** output by the catasb Ansible scripts so you can login using the command line for the remainder of the tutorial.  It will look something like:
```
Comment (jmrodri) [inline docs/getting_started.md]: This paragraph looks good.

File: docs/getting_started.md [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+}
+```
+
+Test out your development environment using [apb-examples](https://github.com/fusor/apb-examples):
+
+```bash
+git clone git@github.com:fusor/apb-examples.git
+cd hello-world-apb
+apb prepare
+docker build -t <docker-user>/hello-world-apb .
```
Comment (jmrodri) [inline docs/getting_started.md]: Will this be replaced with apb build?

File: docs/getting_started.md [hunk 11]
```diff
@@ -0,0 +1,129 @@
...
+docker run -e "OPENSHIFT_TARGET=https://<catasb-host>:<port>" -e "OPENSHIFT_USER=admin" -e "OPENSHIFT_PASS=admin" <docker-user>/hello-world-apb provision
+```
+This should have deployed a hello-world-apb application to your openshift cluster.  Logging on to your cluster at https://<catasb-host>:<port>, you should be able to view the project and click on the deployed application.
```
Comment (jmrodri) [inline docs/getting_started.md]: Let's think more positive, 'This should have deployed...' change to 'This will have deployed ...'

Comment (jmrodri) [reply docs/getting_started.md]: Change 'should be able' to 'will be able'

File: docs/getting_started.md [hunk 12]
```diff
@@ -0,0 +1,129 @@
...
+<a name="first-apb"></a>
+## Creating your first APB
+In this tutorial we'll create a simple PostgreSQL database.
+
+### Using apb init
+Our first task is to create the skeleton for your app using the apb tool.  The command for this is simple:
+```bash
+apb init pg-apb
+```
+At this point you should see the following file structure
```
Comment (jmrodri) [inline docs/getting_started.md]: 'should see' to 'will see'

File: docs/getting_started.md [hunk 13]
```diff
@@ -0,0 +1,129 @@
...
+```
+pg-apb/
+├── apb.yml
+├── Dockerfile
+├── playbooks
+└── roles
+```
+Two files were created at the root directory.  An apb.yml and a Dockerfile.  These are the minimum required for any APB.  For more information, visit the [design doc](https://github.com/fusor/ansible-playbook-bundle/blob/master/docs/design.md#apb-yml).
```
Comment (jmrodri) [inline docs/getting_started.md]: apb.yml to `apb.yml` and Dockerfile to `Dockerfile`.

File: docs/getting_started.md [hunk 14]
```diff
@@ -0,0 +1,129 @@
...
+
+At this point we have a fully formed APB that we can build
+```bash
+cd pg-apb
+apb prepare
+docker build -t <docker-user>/pg-apb .
+docker push <docker-user>/pg-apb
+```
+
+Visiting the OpenShift console UI at https://<catasb-host>:<port> will now display the new Ansible Playbook bundle named pg-apb in the catalog under the "All" tab.
```
Comment (jmrodri) [inline docs/getting_started.md]: Good, in this sentence you used 'will now' which is what I want. :+1:

File: docs/getting_started.md [hunk 15]
```diff
@@ -0,0 +1,129 @@
...
+apb prepare
+docker build -t <docker-user>/pg-apb .
+docker push <docker-user>/pg-apb
+```
+
+Visiting the OpenShift console UI at https://<catasb-host>:<port> will now display the new Ansible Playbook bundle named pg-apb in the catalog under the "All" tab.  
+
+<a name="actions"></a>
+### Actions
+Unfortunately, the brand new APB created in the last section doesn't do very much.  For that, we'll have to add some actions.  The actions supported are [provision](#provision), [deprovision](#deprovision), [bind](#bind), and [unbind](#unbind).  We'll add each of these actions in the following sections
```
Comment (jmrodri) [inline docs/getting_started.md]: I'd remove 'Unfortunately'. The brand new APB, created in the last section, doesn't do very much.

**Review by jmrodri — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (rthallisey) [reply README.md]: This is probably a larger discussion, but should we target kubernetes too? We don't need to have the discussion here, but I think mentioning it as a supported cluster alongside origin is important.

File: docs/getting_started.md (hunk 10 above)
Comment (cfchase) [reply docs/getting_started.md]: yes

**Comment by cfchase** [comment]
Reviewers can click on the "View file" button to actually see the formatted pages.

**Comment by cfchase** [comment]
@jmrodri I think I addressed your concerns.  I've added another commit to add instructions for provision.

File: README.md [hunk 16]
```diff
@@ -1,19 +1,68 @@
...
-Ansible Playbook Bundle (APB) is a lightweight application definition (meta-containers). APB
-has the following features:
+An **Ansible Playbook Bundle (APB)** is a lightweight application definition (meta-container). They are used to define and deploy complex groups of applications, deployment configs, deployments, and services to an [OpenShift Origin](https://github.com/OpenShift/origin) cluster running the [Ansible Service Broker](https://github.com/fusor/ansible-service-broker).  APBs offer more power and simple configuration by leveraging the power of [Ansible](https://www.ansible.com/). APBs have the following features:
 
 * Metadata contains list of required/optional parameters for use during deployment.
 * Leverages existing investment in Ansible Roles / Playbooks.
-* Simple directory with named “action” playbooks and metadata.
-* Developer Tooling to drive guided approach.
+* Actions under a directory with named  **_playbooks_** and metadata defined in **_apb.yml_**.
+* Developer Tooling to drive a guided approach.
```
Comment (jmrodri) [inline README.md]: Make 'Tooling' just 'tooling'.

File: README.md [hunk 17]
```diff
@@ -1,19 +1,68 @@
...
 * Easily modified or extended.
 
+## Installing the ***apb*** tool
+##### Prerequisites
+[Docker](https://www.docker.com/) must be correctly [installed](https://docs.docker.com/engine/installation/) and running on the system.
+
+##### RPM Installation
+Download from [here](????)
+Install using dnf/yum
```
Comment (jmrodri) [inline README.md]: These two lines get rammed together in the formatted version.

```
Download from [here](????)
￼ Install using dnf/yum
```

File: README.md [hunk 18]
```diff
@@ -1,19 +1,68 @@
...
+git clone https://github.com/fusor/ansible-playbook-bundle.git
+```
+Install requirements and run the setup script (requires python)
+```
+sudo pip install -r src/requirements.txt && sudo python setup.py install
+```
+Alternatively you can use [tito](http://github.com/dgoodwin/tito) to install.
+```bash
+tito build --test --rpm -i
+```
```
Comment (jmrodri) [inline README.md:36]: :heart:

**Comment by johnkim76** [comment]
ACK
This looks good to me, and it appears that all the change requests that @jmrodri requested was addressed

File: docs/getting_started.md [hunk 19]
```diff
@@ -0,0 +1,363 @@
...
+docker run \
+    -e "OPENSHIFT_TARGET=https://<catasb-host>:<catasb-port>" \
+    -e "OPENSHIFT_USER=admin" \
+    -e "OPENSHIFT_PASS=admin" \
+    <docker-org>/hello-world-apb provision
+```
+This will deploy a hello-world-apb application to your openshift cluster.  By logging on to your cluster at https://<catasb-host>:<port>, you can view the project and click on the deployed application.
+
+## Creating your first APB
+In this tutorial we'll create an APB for a containerized [hello world application](https://hub.docker.com/r/ansibleplaybookbundle/hello-world/).  We'll work through a basic APB that will mirror the [hello-world-apb](https://github.com/fusor/apb-examples/tree/master/hello-world-apb) in the [apb-examples](https://github.com/fusor/apb-examples) project.
```
Comment (jmrodri) [inline docs/getting_started.md]: add comma (,) after *tutorial*

File: docs/getting_started.md [hunk 20]
```diff
@@ -0,0 +1,363 @@
...
+}
+```
+
+Test out your development environment using [apb-examples](https://github.com/fusor/apb-examples):
+
+```
+git clone git@github.com:fusor/apb-examples.git
+cd hello-world-apb
+apb prepare
+docker build -t <docker-org>/hello-world-apb .
```
Comment (jmrodri) [inline docs/getting_started.md]: Should this be `apb build`? you mention `apb build` later in the document.

File: docs/getting_started.md [hunk 21]
```diff
@@ -0,0 +1,363 @@
...
+At this point we have a fully formed APB that we can build
+```
+cd my-apb
+apb build <docker-org>/my-apb
+docker push <docker-org>/my-apb 
+```
+
+[TODO]: # (We need this to show up in console UI after doing a broker "curl bootstrap" but this doesn't currently show up from the service catalog although it's listed by the broker.)
+
+Visiting the OpenShift console UI at https://<catasb-host>:<port> will now display the new Ansible Playbook bundle named my-apb in the catalog under the **_All_** tab.
```
Comment (jmrodri) [inline docs/getting_started.md]: `s/bundle/Bundle` to be consistent with the rest of the document.

File: docs/getting_started.md [hunk 22]
```diff
@@ -0,0 +1,363 @@
...
+  roles:
+  - role: ansible.kubernetes-modules
+    install_python_requirements: no
+  - role: provision-my-apb
+    playbook_debug: false
+```
+
+`provision.yml` is a new Ansible playbook which will execute on `localhost` and execute the role `provision-my-apb`.  This playbook works on it's local container created by the service broker.  The `ansible.kubernetes-modules` role will allow us to use the [ansible-kubernetes-modules](https://github.com/ansible/ansible-kubernetes-modules) to create our OpenShift resources.  The `provision-my-apb` doesn't exist yet, but that's where all the work will go when deploying our hello-world app.  Let's create it now.
+
+First, create a new file called main.yml inside a roles/provision-my-apb directory.
```
Comment (jmrodri) [inline docs/getting_started.md]: add tick marks around yml and directory:
`main.yml` and `roles/provision-my-apb`.

File: docs/getting_started.md [hunk 23]
```diff
@@ -0,0 +1,363 @@
...
+├── Dockerfile
+├── playbooks
+│   └── provision.yml
+└── roles
+    └── provision-my-apb
+        └── tasks
+            └── main.yml
+```
+
+Rebuilding the APB (`apb build ...`)and running the provision command (`docker run ...`) should output the output from the Ansible playbook with no changes:
```
Comment (jmrodri) [inline docs/getting_started.md]: space after the ) of apb build

File: docs/getting_started.md [hunk 24]
```diff
@@ -0,0 +1,363 @@
...
+      - name: web
+        port: 80
+        target_port: 80
+```
+The `selector` will allow the hello-world service to use the correct pods to include.  The `ports` will take the target port from the pods and expose them as a single port for the service.  If you build the apb and run the provision command, you will see the the new task in the output.
+```
+TASK [provision-my-apb : create hello-world service] ******************
+changed: [localhost]
+```
+You can click on the new service under **_Networking_** in the application on the overview page or under Applications -> Services.  The services's IP address will be shown which you can use to access the load balanced application.  To view the service information from the command line, you can do the following
```
Comment (jmrodri) [inline docs/getting_started.md]: I'd probably make the menu choice italic. _Applications -> Services_

Comment (jmrodri) [reply docs/getting_started.md]: should it be services' or service's?

**Review by jmrodri — CHANGES_REQUESTED** [review]
minor nits. Fix those up. Then I'll approve and merge.

**Comment by cfchase** [comment]
@jmrodri made changes as per your comments

**Review by jmrodri — APPROVED** [review]

**Comment by jmrodri** [comment]
ACK

**Pull request merged**